controller/user_blackbox_test.go

```diff
@@ -0,0 +1,343 @@
+			require.Nil(t, err)
+			// when
+			svc, userCtrl := s.SecuredController(identity)
+			ifModifiedSince := app.ToHTTPTime(identity.User.UpdatedAt)
+			res := test.ShowUserNotModified(t, svc.Context, svc, userCtrl, &ifModifiedSince, nil)
+			// then
+			s.assertResponseHeaders(t, res, identity.User)
+		})
+
+		t.Run("using if-none-match header", func(t *testing.T) {
+			// given
+			identity, err := testsupport.CreateTestIdentityAndUserWithDefaultProviderType(s.DB, "userTestCurrentAuthorizedOK"+uuid.NewV4().String())
+			require.Nil(t, err)
+
+			// when
+			svc, userCtrl := s.SecuredController(identity)
+			ifNoneMatch := app.GenerateEntityTag(identity.User)
+			res := test.ShowUserNotModified(t, svc.Context, svc, userCtrl, nil, &ifNoneMatch)
+			// then
+			s.assertResponseHeaders(t, res, identity.User)
+		})
+	})
+
+	s.T().Run("unauthorized", func(t *testing.T) {
+
+		s.T().Run("missing token", func(t *testing.T) {
+			// given
+			jwtToken := token.New(token.SigningMethodRS256)
+			jwtToken.Claims.(token.MapClaims)["sub"] = uuid.NewV4().String()
+			ctx := jwt.WithJWT(context.Background(), jwtToken)
+			svc, userCtrl := s.UnsecuredController()
+			// when/then
+			test.ShowUserUnauthorized(t, ctx, svc, userCtrl, nil, nil)
+		})
+
+		s.T().Run("ID not a UUID", func(t *testing.T) {
+			// given
+			jwtToken := token.New(token.SigningMethodRS256)
+			jwtToken.Claims.(token.MapClaims)["sub"] = "aa"
+			ctx := jwt.WithJWT(context.Background(), jwtToken)
+			svc, userCtrl := s.UnsecuredController()
+			// when/then
+			test.ShowUserUnauthorized(t, ctx, svc, userCtrl, nil, nil)
+		})
+
+		s.T().Run("token without identity", func(t *testing.T) {
+			// given
+			jwtToken := token.New(token.SigningMethodRS256)
+			ctx := jwt.WithJWT(context.Background(), jwtToken)
+			svc, userCtrl := s.UnsecuredController()
+			// when/then
+			test.ShowUserUnauthorized(t, ctx, svc, userCtrl, nil, nil)
+		})
+
+		s.T().Run("deprovisionned user", func(t *testing.T) {
+			// given
+			identity, err := testsupport.CreateDeprovisionedTestIdentityAndUser(s.DB, "TestShowDeprovisionedUserFails"+uuid.NewV4().String())
+			require.NoError(t, err)
+			svc, userCtrl := s.SecuredController(identity)
+			// when
+			rw, _ := test.ShowUserUnauthorized(t, svc.Context, svc, userCtrl, nil, nil)
+			// then
+			assert.Equal(t, "DEPROVISIONED description=\"Account has been deprovisioned\"", rw.Header().Get("WWW-Authenticate"))
+			assert.Contains(t, "WWW-Authenticate", rw.Header().Get("Access-Control-Expose-Headers"))
+		})
+
+		s.T().Run("", func(t *testing.T) {
+
+		})
+
+		s.T().Run("", func(t *testing.T) {
+
+		})
+
+	})
+}
+
+func (s *UserControllerTestSuite) TestListUserSpaces() {
+
+	spacesResourceType := "spaces"
+
+	s.T().Run("ok", func(t *testing.T) {
+
+		t.Run("role on no space", func(t *testing.T) {
+			// given
+			g := s.NewTestGraph()
```

Comment:
I think you can use `s.Graph` directly here, because we are creating it beforeEach test. https://github.com/fabric8-services/fabric8-auth/blob/master/gormtestsupport/db_test_suite.go#L72

Reply:
I'd rather create a new TestGraph for each subtest because in the latest commits of this PR, I need to pass the current `*testing.T` as an argument, because in case of assertion failure, the test panics (that's a side-effect of using sub tests)

Reply:
It's one of the reason I don't like subtests :) It looks nicer in test reports but doesn't provide good isolation between separate tests.

Reply:
I’m not saying that nicer reports is the only advantage of subtests. I just pointed that there are some disadvantages too. Like lack of proper test isolation.

Reply:
yes, I agree with you for the lack of test isolation, and this is annoying. I'll see if there's something we can do about it. For now, a good way to work-around this limitation is by making sure that test data do not conflict with each other...